src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,196 @@
+import userStorage from '../../lib/gundb/UserStorage'
+import FeedContactItem from './FeedContactItem'
+const Contacts = promisify(contacts.getAll)
+
+const WhoContent = ({ styles, setContact, error, text, value, next, state, showNext, setValue }) => {
+  const [contacts, setContacts] = React.useState([])
+  const [initialList, setInitalList] = React.useState(contacts)
+  const [recentFeedItems, setRecentFeedItems] = React.useState([])
+  const [recentlyUsedList, setRecentlyUsedList] = React.useState([])
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      }
+      Contacts().then((contacts, err) => {
+        if (err === 'denied') {
+          console.warn('permissions denied')
+        } else {
+          const sortContacts = orderBy(contacts, ['givenName'])
+          setContacts(sortContacts)
+          setInitalList(sortContacts)
+        }
+      })
+    }
+  }, [])
+
+  const filterContacts = useCallback(
+    memoize(searchQuery => {
+      const query = searchQuery.toLocaleLowerCase()
+
+      return initialList.filter(({ givenName, familyName, phoneNumbers }) => {
+        const [phoneNumber] = map(phoneNumbers, 'number')
+
+        return [givenName, familyName, phoneNumber].some(field => (field || '').toLocaleLowerCase().includes(query))
+      })
+    }),
+    [initialList]
+  )
+
+  const handleSearch = useCallback(
+    query => {
+      if (state) {
+        setValue(query)
+      }
+
+      setContacts(filterContacts(query))
+    },
+    [setContacts, setValue, filterContacts]
+  )
+
+  useEffect(() => {
+    if (contacts.length === 0) {
+      showNext(true)
+    } else {
+      showNext(false)
+    }
+  }, [contacts])
+
+  useEffect(() => {
+    findRecentlyUsed()
+  }, [contacts, recentFeedItems])
+
+  const findRecentlyUsed = () => {
+    let matches = []
+    if (recentFeedItems.length > 0 && contacts.length > 0) {
+      recentFeedItems.forEach(feedItem => {
+        matches.push(
+          contacts.filter(({ phoneNumbers }) => {
+            const [contactNumber] = map(phoneNumbers, 'number')
+            const { phoneNumber } = feedItem.data
+            return contactNumber && phoneNumber === contactNumber
+          })
+        )
+      })
+    }
+
+    const recentlyContacts = uniq(matches.flat())
+
+    setRecentlyUsedList(recentlyContacts)
+  }
+
+  const renderItem = useCallback(
```

Comment:
1. memoize is needed here to, as useCallback memoizes the renderItem function only, you need also memoize render functions  it returns
2. function doesn't renders actually, it **returns renderer**, so it should be renamed
3. renderItem(true) looks unspecified. what `true` is ? i propose to add `mode` param with enum type 'horizontalMode' | 'verticalMode'

```javascript
const getItemRenderer = useCallback(
     memoize((mode = 'verticalMode')  => ({ item, index }) => {
       return <FeedContactItem contact={item} selectContact={setContact} horizontalMode={mode === 'horizontalMode'} index={index} />
     }),
     [setContact]
   )
```